src/launcher_gui.js

```diff
@@ -21,89 +26,123 @@ app.on('second-instance', () => {
 });
 
 app.prependListener('ready', () => {
-	const { wizard } = require('./launcher_wizard');
-
-	const display = electron.screen.getPrimaryDisplay();
-	const sWidth = display.workAreaSize.width;
-	const width = 800;
-	const height = process.platform === 'win32' ? 418 : 380 + 8;
-
-	let windowOpts = {
-		x: (sWidth - width) / 2,
-		// y: (sHeight - height) / 2,
-		y: 100,
-		width: width,
-		height: height,
-		show: false,
-		icon: `${__dirname}/renderer/spring-icon.png`,
-		webPreferences: {
-			nodeIntegration: true,
-			enableRemoteModule: true,
-		},
-	};
-	windowOpts.resizable = false;
-	Menu.setApplicationMenu(null);
-	mainWindow = new BrowserWindow(windowOpts);
-
-	mainWindow.loadFile(`${__dirname}/renderer/index.html`);
-	if (isDev) {
-		// mainWindow.webContents.openDevTools();
-	}
+	// TODO: is there a better way to get yargs' positional args ? Did not find
+	// it ...
+	if (argv._[0]) {
+		// If a replay file was passed as argument to spring-launcher, then we
+		// don't open the GUI, we just start the replay.
+
+		// TODO: when we support downloading assets (map/game/engine) we might
+		// need to show the GUI anyway, to provide feedback on the
+		// downloads/ask for user confirmation.
+		const { launcher } = require('./engine_launcher');
+
+		const replayPath = decodeURI(url.parse(argv._[0]).pathname);
+
+		const engineName = replayPath.split('_').pop().slice(0, -5).toLowerCase();
+		const enginePath = join(
+			springPlatform.writePath,
+			'engine',
+			engineName,
+			springPlatform.springBin
+		);
+
+		// For the moment, we'll just check if the engine exists. Spring will
+		// give error messages if the map or the game is not found.
+		if (!fs.existsSync(enginePath)) {
+			dialog.showMessageBoxSync(
+				null, {message: `Engine not found: ${engineName}`}
+			);
+		}
 
-	mainWindow.on('closed', () => {
-		mainWindow = null;
-		app.quit();
-	});
-
-	tray = new Tray(`${__dirname}/renderer/spring-icon.png`);
-	var template = [
-		// TODO: About dialog that shows URL, author, version, etc.
-		// {
-		//   role: 'about',
-		//   click: () => {
-		//     log.info("About clicked");
-		//   }
-		// },
-		{
-			// TODO: Proper "show/hide"
-			label: 'Toggle hide',
-			click: () => {
-				if (mainWindow.isVisible()) {
-					//menuItem.label = "Show";
-					mainWindow.hide();
-				} else {
-					mainWindow.show();
-					//menuItem.label = "Hide";
+		launcher.launchSpring(enginePath, [replayPath]);
+	} else {
```

Comment:
avoid long, mostly unrelated if-else blocks

prefer to separate these blocks into functions:
```js
if (headlessMode) {
    launchHeadless();
} else {
    setupGUI();
}
```